File: solutions/csharp/CoffeeVendingMachine/CoffeeVendingMachineDemo.cs
namespace CoffeeVendingMachine;

public class CoffeeVendingMachineDemo
{
    public static void Main(string[] args)
    {
        CoffeeMachine coffeeMachine = CoffeeMachine.GetInstance();

        // Display coffee menu
        coffeeMachine.DisplayMenu();

        // Simulate user requests
        Coffee espresso = coffeeMachine.SelectCoffee("Espresso");
        coffeeMachine.DispenseCoffee(espresso, new Payment(3.0));

        Coffee cappuccino = coffeeMachine.SelectCoffee("Cappuccino");
        coffeeMachine.DispenseCoffee(cappuccino, new Payment(3.5));


        Coffee latte = coffeeMachine.SelectCoffee("Latte");
        coffeeMachine.DispenseCoffee(latte, new Payment(4.0));
    }
}

File: solutions/golang/coffeevendingmachine/coffee_vending_machine_demo.go
package coffeevendingmachine

import (
	"fmt"
)

func Run() {
	coffeeMachine := GetCoffeeMachine()

	// Display coffee menu
	coffeeMachine.DisplayMenu()

	// Simulate user requests
	coffeeTypes := []string{"Espresso", "Cappuccino", "Latte"}
	payments := []float64{3.0, 3.5, 4.0}

	for i, coffeeType := range coffeeTypes {
		selectedCoffee := coffeeMachine.SelectCoffee(coffeeType)
		payment := NewPayment(payments[i])

		fmt.Printf("\nOrdering %s...\n", coffeeType)
		if err := coffeeMachine.DispenseCoffee(selectedCoffee, payment); err != nil {
			fmt.Printf("Error: %v\n", err)
		}
	}
}
